src/micm/micm.cpp

```diff
@@ -167,21 +232,55 @@ namespace musica
     return micm->GetSpeciesProperty<bool>(species_name_str, property_name_str, error);
   }
 
-  void MICM::Create(const std::string &config_path, Error *error)
+  void MICM::CreateRosenbrock(const std::string &config_path, Error *error)
   {
     try
     {
       micm::SolverConfig<> solver_config;
       solver_config.ReadAndParse(std::filesystem::path(config_path));
       solver_parameters_ = std::make_unique<micm::SolverParameters>(solver_config.GetSolverParams());
 
-      solver_ = std::make_unique<Rosenbrock>(micm::CpuSolverBuilder<micm::RosenbrockSolverParameters>(
-                                                 micm::RosenbrockSolverParameters::ThreeStageRosenbrockParameters())
-                                                 .SetSystem(solver_parameters_->system_)
-                                                 .SetReactions(solver_parameters_->processes_)
-                                                 .SetNumberOfGridCells(NUM_GRID_CELLS)
-                                                 .SetIgnoreUnusedSpecies(true)
-                                                 .Build());
+      rosenbrock_ = std::make_unique<Rosenbrock>(
+          micm::SolverBuilder<
+              micm::RosenbrockSolverParameters,
+              micm::VectorMatrix<double, MICM_VECTOR_MATRIX_SIZE>,
+              micm::SparseMatrix<double, micm::SparseMatrixVectorOrdering<MICM_VECTOR_MATRIX_SIZE>>,
+              micm::ProcessSet,
+              micm::LinearSolver<
+                  micm::SparseMatrix<double, micm::SparseMatrixVectorOrdering<MICM_VECTOR_MATRIX_SIZE>>,
+                  micm::LuDecomposition>>(micm::RosenbrockSolverParameters::ThreeStageRosenbrockParameters())
```

Comment:
maybe we should make something similar to `micm::CpuSolverBuilder` but make it be `micm::VectorizedCpuSolverBuilder` which is templated on the on vector size. Then it would be able to do all of this gross stuff for us. As an added benefit we should be able to add some alias types inside all the solver builder aliases that give us the type of the solver so that inside the header file we would only need to do something like

```
using builderType.= micm::CpuSolverBuilder;
using solverType = builderType::RosenbrockType;
```

Reply:
I like this idea. I created [an issue](https://github.com/NCAR/micm/issues/602) for this in MICM. I think this can be a separate issue because alias-declaration for builder type should be done first at MICM.